pkg/handlers/signup/provisioner/handlers.go

```diff
@@ -0,0 +1,164 @@
+package provisioner
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+	core "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/konflux-ci/workspace-manager/pkg/api/v1alpha1"
+)
+
+const (
+	UserEmailAnnotation string = "konflux-ci.dev/requester-email"
+	UserIdAnnotation    string = "konflux-ci.dev/requester-user-id"
+)
+
+func normalizeEmail(email string) string {
+	ret := strings.Replace(email, "@", "-", -1)
+	ret = strings.Replace(ret, "+", "-", -1)
```

Comment:
Any concern that users with different emails will get the same normalized email? e.g. user+1@example.com and user-1@example.com (or users with very long emails that start the same)

Reply:
I agree that this may be a concern long term. The alternatives I can think of to avoid collisions will lead to ugly namespace names and/or break idempotency, like appending a kubernetes generated suffix or using IdP's user Id.

What if we allow the user to define the namespace name instead? We'll break idempotency, but we'll also avoid collisions. We should also implement a quota mechanism, that can be very simple for now: we can store the user email (or email hash to be RFC1123 compliant) in an annotation/label. In the create endpoint we can check that the user has not already created a namespace by checking this annotation/label on existing namespaces.